Plugins/UnityServerPlugin/StreamingUnityServerPlugin.cpp

```diff
@@ -408,16 +553,61 @@ extern "C" UnityRenderingEvent UNITY_INTERFACE_EXPORT UNITY_INTERFACE_API GetRen
 	return OnEncode;
 }
 
-extern "C" __declspec(dllexport) void SetInputDataCallback(void(__stdcall*onInputUpdate)(const char *msg))
+extern "C" __declspec(dllexport) void SetCallbackMap(StrParamFuncType onInputUpdate,

```

Comment:
this basically has to be gross afaik, because if we pass a struct the marshalling doesn't seem to work properly